src/SprykerEco/Yves/Braintree/Model/Mapper/PaypalResponse/PaypalResponseMapper.php

```diff
@@ -0,0 +1,233 @@
+        $this->countryClient = $countryClient;
+        $this->moneyPlugin = $moneyPlugin;
+    }
+
+    /**
+     * @param array $payload
+     *
+     * @return \Generated\Shared\Transfer\PaypalExpressSuccessResponseTransfer
+     */
+    public function mapSuccessResponseToPaypalExpressSuccessResponseTransfer(array $payload): PaypalExpressSuccessResponseTransfer
+    {
+        $transfer = new PaypalExpressSuccessResponseTransfer();
+
+        $transfer->setNonce($payload[static::KEY_NONCE] ?? null);
+        $transfer->setEmail($payload[static::KEY_DETAILS][static::KEY_EMAIL] ?? null);
+        $transfer->setFirstName($payload[static::KEY_DETAILS][static::KEY_FIRST_NAME] ?? null);
+        $transfer->setLastName($payload[static::KEY_DETAILS][static::KEY_LAST_NAME] ?? null);
+        $transfer->setPayerId($payload[static::KEY_DETAILS][static::KEY_PAYER_ID] ?? null);
+        $transfer->setRecipientName($payload[static::KEY_DETAILS][static::KEY_SHIPPING_ADDRESS][static::KEY_RECIPIENT_NAME] ?? null);
+        $transfer->setLine1($payload[static::KEY_DETAILS][static::KEY_SHIPPING_ADDRESS][static::KEY_LINE1] ?? null);
+        $transfer->setCity($payload[static::KEY_DETAILS][static::KEY_SHIPPING_ADDRESS][static::KEY_CITY] ?? null);
+        $transfer->setState($payload[static::KEY_DETAILS][static::KEY_SHIPPING_ADDRESS][static::KEY_STATE] ?? null);
+        $transfer->setPostalCode($payload[static::KEY_DETAILS][static::KEY_SHIPPING_ADDRESS][static::KEY_POSTAL_CODE] ?? null);
+        $transfer->setCountryCode($payload[static::KEY_DETAILS][static::KEY_SHIPPING_ADDRESS][static::KEY_COUNTRY_CODE] ?? null);
+        $transfer->setCurrency($payload[static::KEY_CURRENCY] ?? null);
+        $transfer->setAmount($payload[static::KEY_AMOUNT] ?? null);
+
+        return $transfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PaypalExpressSuccessResponseTransfer $paypalExpressSuccessResponseTransfer
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function mapPaypalExpressSuccessResponseTransferToQuoteTransfer(
+        PaypalExpressSuccessResponseTransfer $paypalExpressSuccessResponseTransfer,
+        QuoteTransfer $quoteTransfer
+    ): QuoteTransfer {
+        $this->addCustomer($quoteTransfer, $paypalExpressSuccessResponseTransfer);
+        $this->addShippingAddress($quoteTransfer, $paypalExpressSuccessResponseTransfer);
+        $this->addBillingTransfer($quoteTransfer, $paypalExpressSuccessResponseTransfer);
+        $this->addPaymentTransfer($quoteTransfer, $paypalExpressSuccessResponseTransfer);
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\PaypalExpressSuccessResponseTransfer $paypalExpressSuccessResponseTransfer
+     *
+     * @return void
+     */
+    protected function addPaymentTransfer(
```

Comment:
`add*` methods here are modifying `QuoteTransfer`. In this case we should return `QuoteTransfer` back to make it more obvious and avoid "silence" object modifying.